app_context/basic_test.go

```diff
@@ -6,6 +6,34 @@ import (
 	"testing"
 )
 
+func TestTiltEnv(t *testing.T) {
+	os.Unsetenv("TILT_ENVIRONMENT")
+	app_ctx, err := NewAppContext("basic_test")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if tiltEnv := app_ctx.TiltEnv(); tiltEnv != "development" {
+		t.Errorf("environment is not development: %s", tiltEnv)
+	}
+
+	os.Setenv("TILT_ENVIRONMENT", "testing")
+
+	app_ctx, err = NewAppContext("basic_test")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if tiltEnv := app_ctx.TiltEnv(); tiltEnv != "testing" {
+		t.Errorf("environment is not testing: %s", tiltEnv)
+	}
+
+	os.Setenv("TILT_ENVIRONMENT", "derp")
+
+	app_ctx, err = NewAppContext("basic_test")
+	if err == nil {
+		t.Error("app context should have failed")
+	}
```

Comment:
@comstud This very last test seems fine to me, as it wants to make sure err is at least set.  However, the overall test fails, and I'm wondering why
```
% go test ./app_context
2017/01/05 15:34:59 Unknown TILT_ENVIRONMENT: derp
FAIL	github.com/tilteng/go-app-context/app_context	0.021s
```